### PR TITLE
Control logic modified to work with FRC robot code and with data validation

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/org/usfirst/frc/team4859/robot/Robot.java [hunk 1]
```diff
@@ -82,108 +83,32 @@ public void disabledPeriodic() {
...
-		 * = new MyAutoCommand(); break; case "Default Auto": default:
-		 * autonomousCommand = new ExampleCommand(); break; }
-		 
 
-		// schedule the autonomous command (example)
-		if (m_autonomousCommand != null) {
-			m_autonomousCommand.start();
-		}
-		*/
 		String robotPos = SmartDashboard.getString("Robot Start Pos (L,R, or C)", "Non Received");
```
Comment (daheinz) [inline src/org/usfirst/frc/team4859/robot/Robot.java:87]: Should we default to one of the allowed values?

File: src/org/usfirst/frc/team4859/robot/Robot.java [hunk 2]
```diff
@@ -82,108 +83,32 @@ public void disabledPeriodic() {
...
+		
+		String targetScale = SmartDashboard.getString("Target", "Y");
+		
+		RobotMap.delayInSeconds = SmartDashboard.getNumber("Auton Delay", 0);
+		
+		String gameData = DriverStation.getInstance().getGameSpecificMessage();
+		
+	//MAKE SURE GAME DATA IS GOOD
+		// Please don't hurt me for this
+		if (!gameData.equalsIgnoreCase("RRR") || !gameData.equalsIgnoreCase("LLL") || !gameData.equalsIgnoreCase("RLR") || !gameData.equalsIgnoreCase("LRL") || !robotPos.equalsIgnoreCase("L") || !robotPos.equalsIgnoreCase("C") || !robotPos.equalsIgnoreCase("R") || (RobotMap.delayInSeconds >= 0 && RobotMap.delayInSeconds <= 15)) {
```
Comment (daheinz) [inline src/org/usfirst/frc/team4859/robot/Robot.java:98]: Will this be called  on the transition to Autonomous Enabled as recommended here : t https://wpilib.screenstepslive.com/s/currentCS/m/getting_started/l/826278-2018-game-data-details#initial-states

**Review by daheinz — APPROVED** [review]

File: src/org/usfirst/frc/team4859/robot/Robot.java (hunk 2 above)
Comment (daheinz) [reply src/org/usfirst/frc/team4859/robot/Robot.java:98]: consider a regrex

Comment (morstad) [reply src/org/usfirst/frc/team4859/robot/Robot.java:98]: @Chaube97 @ecockr38 created quick example for regexp in https://github.com/Byron-MN-FRC/Java_Education/blob/nancy-branch/src/testflow/AutonSelector.java 

```
		boolean valid = Pattern.matches("[LR]{3}", gameData.toUpperCase());

```
The main is a good way to validate expression once have it.   There are many sites that will help you create/test regrexp as well..  For example : https://www.regexplanet.com/advanced/java/index.html

**Review by morstad — APPROVED** [review]

**Comment by morstad** [comment]
@Chaube97 Left actual merge to you incase wanted to tweak input validation further

**Pull request merged**